### PR TITLE
prevent validating actions that have errored.

### PR DESCRIPTION
**Pull request by Eruanion**
## Related issue

Closes: #168 

## Description
If the solution fails to port due to various reasons we are still validating the results which creates a spike of false validation errors.

## Supplemental testing
All unit test pass, plus localized testing for ensuring logic works as intended.

## Additional context
Please provide any additional information related to this PR

---
*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

**Review by ShanshanHe — APPROVED** [review]
LGTM

File: src/CTA.Rules.Update/CodeReplacer.cs [hunk 1]
```diff
@@ -271,6 +275,9 @@ public List<GenericActionExecution> ValidateActions(List<GenericActionExecution>
                 {
                     var actionValidationException = new ActionValidationException(action.Key, action.Name);
                     action.InvalidExecutions = 1;
+                    var originalAction = actions.FirstOrDefault(a => a.GetHashCode() == action.GetHashCode());
+                    if(originalAction != null) 
+                        originalAction.InvalidExecutions = 1;
```
Comment (marknfawaz) [inline src/CTA.Rules.Update/CodeReplacer.cs:280]: Why do we need this?

**Review by marknfawaz — CHANGES_REQUESTED** [review]

File: src/CTA.Rules.Update/CodeReplacer.cs (hunk 1 above)
Comment (Eruanion) [reply src/CTA.Rules.Update/CodeReplacer.cs:280]: It was modifying the original actions list here to set the invalid executions if needed. Iterating over a new reduced list that doesn't get returned I need to access original list that gets returned to modify the corresponding element.

Comment (Eruanion) [reply src/CTA.Rules.Update/CodeReplacer.cs]: Sorry this was my mistake, it doe sindeed pass by reference and this code is superfluous and not needed, will remove.

**Review by marknfawaz — APPROVED** [review]

**Pull request merged**